src/components/ScriptExplorer/ScriptEntry.jsx

```diff
@@ -228,7 +228,7 @@ class ScriptEntry extends React.Component {
     } else {
       this.setState({
         fetchedUTXOs: false,
-        fetchUTXOsError: "Failed to fetch UTXOs.",
+        fetchUTXOsError: "This address has a zero balance.",
```

Comment:
This `else` block will hit if the error statement is not blank which means it could be for a real reason other than no balance. In fact it looks like line 209 does the same thing you're doing here but if the balance retrieved is actually zero, so it's possible that something else did go wrong for real here. Perhaps it might be more useful to set this to the `fetchUTXOsError` value or at least log it to the console.

Reply:
Good point. I changed the object returned by `fetchUTXOs` so that the error can be logged. Also if no UTXOs where found the "This address has a zero balance." is triggered on line 209.